app/models/big_blue_button_conference.rb

```diff
@@ -81,6 +81,49 @@ def recordings
     end
   end
 
+  def delete_all_recordings
+    fetch_recordings.map do |recording|
+      delete_recording recording[:recordID]
+    end
+  end
+
+  def close
+    end_meeting
+    super
+  end
+
+  def destroy
+    end_meeting
+    delete_all_recordings
+    super
+  end
+
+  set_policy do
```

Comment:
Wit code duplication, do you mean the close/destroy methods?

Regarding to the set_policy, yeah I know what you mean, I tried not to move anything in the superclass because I think you use that class for other conference systems.
In fact the only change required is to ignore long_running for bigbluebutton because it should be possible to complete any meeting.

    given { |user, session| context.grants_right?(user, session, :manage_content) && active? }
    can :close